nomad/grid-intensity-exporter.nomad

```diff
@@ -0,0 +1,44 @@
+
+job "grid-intensity-exporter" {
+
+  # The "datacenters" parameter specifies the list of datacenters which should
+  # be considered when placing this task. This must be provided.
+  datacenters = ["dc1"]
+
+  # system jobs run on every client node in a cluster, which is what we
+  # want, but this bug here means that running this job as a system on
+  # means that network ports aren't allocated properly
+  # https://github.com/hashicorp/nomad/issues/8934
+  # until it's resolved, we need to run this as a service instead
+  type = "service"
```

Comment:
Would you mind checking if this bug still exists, and creating an issue to update the jobfile if so?

I was reading the release notes for Nomad recently and I think they did some work around how system jobs are run.

Reply:
Actually Chris I'm doubting this now.

For monitoring agents like scaphandre we'd definitely want a system job to collect metrics from all nodes.

But the exporter is fetching data from the external provider so it's more efficient to run a service with a single instance that prometheus can scrape.

I'm going to merge this but I'll create a follow up PR to update the comments in the job and we can discuss there.

Reply:
hey @rossf7  I think that's fair comment. I had initially been thinking in terms of the nodes running this based on where they geographically, and I was thinking of [this image from the nomad 1.3 release notes](https://www.hashicorp.com/blog/nomad-1-3-adds-native-service-discovery-and-edge-workload-support) initially.

![single nomad cluster, clients in multiple zones](https://www.datocms-assets.com/2885/1649346368-global-nomad-federation.png?fit=max&fm=webp&q=80&w=2500)

But you're right, the data **is** from an external provider in all the cases we have so far, rather than from the sites themselves.

Until we have data from data sources local to the node, like the [solar protocol](http://solarprotocol.net/) does, you're right. I'm happy to follow your lead. 

Would you please make an issue to capture that use case though? This [paper here from Facebook about carbon explorer](https://github.com/facebookresearch/CarbonExplorer/), basically explains how they do this node-local data source stuff already, and it would be good to support this use case in future.

Reply:
Ah got it. Right now I'd handle this by having an instance of the exporter per DC in the nomad cluster.

Each instance can be configured with the region the DC is located in.

You're right though that lots of hosting providers have metadata APIs that can be used to find where a node is. It would be very cool to build auto discovery into the exporter.

Thanks for the links Solar protocol looks very interesting as does the facebook tool. I'll create a follow up issue for this. 

For now this little PR updates the nomad job. https://github.com/thegreenwebfoundation/grid-intensity-go/pull/37